### PR TITLE
[Enterprise Search][Application] Modularize ent-search plugin.

### PR DESCRIPTION
**Pull request by afoucret**
Everything is in the PR title.

Feel free to merge it into the ent-search-branch or to wait that it have landed into main.

### CODE REVIEW

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java [hunk 1]
```diff
@@ -7,8 +7,6 @@
 
 package org.elasticsearch.xpack.application.search;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
```
Comment (afoucret) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/search/SearchApplicationIndexService.java:11]: Using `org.elasticsearch.logging.*` instead so we have one less module to require (`org.apache.logging.log4j`)

File: x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java [hunk 2]
```diff
@@ -7,8 +7,6 @@
 
 package org.elasticsearch.xpack.application;
 
-import org.apache.logging.log4j.LogManager;
-import org.apache.logging.log4j.Logger;
```
Comment (afoucret) [inline x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/EnterpriseSearch.java:11]: Using org.elasticsearch.logging.* instead so we have one less module to require (org.apache.logging.log4j)

File: x-pack/plugin/ent-search/src/main/java/module-info.java [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    requires org.elasticsearch.logging;
+    requires org.elasticsearch.server;
+    requires org.elasticsearch.xcontent;
+    requires org.elasticsearch.xcore;
+
+    exports org.elasticsearch.xpack.application.analytics;
+    exports org.elasticsearch.xpack.application.analytics.action;
+
+    exports org.elasticsearch.xpack.application.search;
+    exports org.elasticsearch.xpack.application.search.action;
```
Comment (afoucret) [inline x-pack/plugin/ent-search/src/main/java/module-info.java:21]: We could be more strict on which module can use our code but I do not see a huge benefits in doing it like in security.

**Review by jimczi — APPROVED** [review]
LGTM, let's merge in the branch and I'll ask the es team to review this specific commit in our main PR.

**Comment by afoucret** [comment]
@jimczi I have a build failure that I am unable to explain. I would bet it is flakiness.
Do we merge?

**Comment by afoucret** [comment]
Super. TIL about the elasticmachine.
Is there a list of the available commands somewhere?

**Pull request merged**